File: 351004/Mukhin/Lab1/restApi/publisher/src/main/java/com/example/restApi/controllers/MarkerController.java
package com.example.restApi.controllers;

import com.example.restApi.dto.request.MarkerRequestTo;
import com.example.restApi.dto.response.MarkerResponseTo;
import com.example.restApi.services.MarkerService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/markers")
public class MarkerController {

    private final MarkerService markerService;

    public MarkerController(MarkerService markerService) {
        this.markerService = markerService;
    }

    @GetMapping
    public ResponseEntity<List<MarkerResponseTo>> getAll(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "100") int size,
            @RequestParam(defaultValue = "id") String sort) {
        return ResponseEntity.ok(markerService.getAll(page, size, sort).getContent());
    }

    @GetMapping("/{id}")
    public ResponseEntity<MarkerResponseTo> getById(@PathVariable Long id) {
        return ResponseEntity.ok(markerService.getById(id));
    }

    @PostMapping
    public ResponseEntity<MarkerResponseTo> create(@Valid @RequestBody MarkerRequestTo request) {
        MarkerResponseTo response = markerService.create(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    @PutMapping("/{id}")
    public ResponseEntity<MarkerResponseTo> update(@PathVariable Long id,
                                                   @Valid @RequestBody MarkerRequestTo request) {
        return ResponseEntity.ok(markerService.update(id, request));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        markerService.delete(id);
        return ResponseEntity.noContent().build();
    }
}

File: 351003/Guzaev/publisher/app/controllers/marker_controller.py
from fastapi import APIRouter, status, Response
from typing import List
from dtos.marker_dto import MarkerRequestTo, MarkerResponseTo
from services.marker_service import MarkerService

router = APIRouter(prefix="/api/v1.0/markers", tags=["markers"])
service = MarkerService()

@router.post("", status_code=status.HTTP_201_CREATED, response_model=MarkerResponseTo)
def create_marker(dto: MarkerRequestTo):
    return service.create(dto)

@router.get("", response_model=List[MarkerResponseTo])
def get_markers():
    return service.get_all()

@router.get("/{id}", response_model=MarkerResponseTo)
def get_marker(id: int):
    return service.get_by_id(id)

@router.put("/{id}", response_model=MarkerResponseTo)
def update_marker(id: int, dto: MarkerRequestTo):
    return service.update(id, dto)

@router.delete("/{id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_marker(id: int):
    service.delete(id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)
